Description: Don't link QtWebkit with --gc-sections on powerpc and s390
 This works around a binutils bug that results in a segfault on powerpc,
 an internal link error on s390, and OOM condition on mipsel.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Author: Modestas Vainius <modax@debian.org>
Origin: vendor
Forwarded: no
Last-Update: 2010-05-31 (mipsel added)

---
 src/3rdparty/webkit/Source/WebCore/WebCore.pri |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/3rdparty/webkit/Source/WebCore/WebCore.pri
+++ b/src/3rdparty/webkit/Source/WebCore/WebCore.pri
@@ -324,7 +324,7 @@ mac {
 }
 
 unix:!mac:*-g++*:QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
-unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+unix:!mac:*-g++*:!equals(QT_ARCH, powerpc):!equals(QT_ARCH, s390):!equals(QT_ARCH, mips):!equals(QT_ARCH, arm):QMAKE_LFLAGS += -Wl,--gc-sections
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 unix|win32-g++*:QMAKE_PKGCONFIG_REQUIRES = QtCore QtGui QtNetwork
